sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,650 @@
+     *                      A subclass of RuntimeException, IllegalArgumentException is thrown when username or password is null or empty.
+     *                      A subclass of RuntimeException, NullPointerException is thrown when session id returned from server is null.
+     * @return				Session ID. Will be null if the call fails.
+     */
+    public String loginUser( String username, String password) throws IOException, RuntimeException
+    {
+        if(username == null || username.isEmpty() || password == null || password.isEmpty())
+            throw new IllegalArgumentException("invalid input of username or password");
+
+        final String url = URL_AUTH + "/" + URL_LOGIN;
+        final JSONObject user = new JSONObject();
+        try {
+            user.put("username", username);
+            user.put("password", password);
+        }catch(JSONException JS){
+            throw new RuntimeException("loginUser failed to create the content for the request");
+        }
+        Map<String, String> response = request(url, user, null, HTTP_METHOD_POST);
+        // if response code is not 200 (OK), the login was incorrect
+        int result = checkResponseCode(response.get(RESPONSE_CODE), "login");
+        if(result != 0)
+            throw new IOException("failed to log in, response code is:" + response.get(RESPONSE_CODE));
+        String session_id = response.get("session-id");
+        if (session_id == null)
+            Log.w(TAG, "loginUser returns a null session id");
+
+        return session_id;
+    }
+
+
+    /**
+     * Logout the currently logged in user.
+     *
+     * @param sessionID		The sessionID of the user to logout. Cannot be empty.
+     * @throws IOException, IllegalArgumentException
+     *                      IOException is thrown from request() method when the the response content(inputStream) has errors,
+     *                      IllegalArgumentException is thrown when session id is null or empty.
+     * @return				Whether or not the logout finished successfully.
+     */
+    public boolean logoutCurrentUser(String sessionID) throws IOException, IllegalArgumentException
+    {
+        if(sessionID == null || sessionID.isEmpty())
+            throw new IllegalArgumentException("invalid input of session ID");
+
+        final String url = URL_AUTH + "/" + URL_LOGOUT;
+        Map<String, String> response = request(url, null, sessionID, HTTP_METHOD_POST);
+        // if response code is not 200 (OK), the logout was incorrect
+        String responseCode = response.get(RESPONSE_CODE);
+        int result = checkResponseCode(response.get(RESPONSE_CODE), "logout");
+
+        return (result == 0);
+
+    }
+
+    /**
+     * Create a new sensor in the commonsense backend
+     *
+     * Each sensor in commonsense is uniquely identified by a name and device type combination.
+     * If this combination already exists in the commonsense backend this call will fail.
```

Comment:
this is wrong "Each sensor in commonsense is uniquely identified by a name and device type combination.
If this combination already exists in the commonsense backend this call will fail."